source/module_hamilt_general/module_surchem/minimize_cg.cpp

```diff
@@ -184,9 +184,8 @@ void surchem::Leps2(const UnitCell& ucell,
         grad_phi[ir].y *= epsilon[ir];
         grad_phi[ir].z *= epsilon[ir];
     }
-
-    double *lp_real = new double[rho_basis->nrxx];
-    ModuleBase::GlobalFunc::ZEROS(lp_real, rho_basis->nrxx);
+    std::vector<double> lp_real(rho_basis->nrxx,0);
+    ModuleBase::GlobalFunc::ZEROS(&lp_real[0], rho_basis->nrxx);
```

Comment:
note that you have already zero-initialized the `std::vector`, why do you still call `ModuleBase:GlobalFunc::ZEROS`?